[#4060] Align completeness between old Aggregate and new Entity description (#4562)
AxonFramework/AxonFramework: docs/reference-guide/modules/commands/pages/command-handlers.adoc

Reviewer: I think this page requires some changes now as well. With the new files introduced, this page has a lot of duplicates. 

We originally **only** had this page for Command Handling description in AF5, to safe on time. Now that we have time, I'd like to "unfold" this page into distinct pages again. References to one another are fine of course, but brunt of the description should live in the new files you've added, not here.

Reviewer: I'd prefer this example to be a separate chapter as well, actually 👼 
It is **the** new way of handling commands and entities, allowing users to abstract out command handling logic from an entity object while still using annotations. Added, it is **the** way how to do Vertical Slice Architecture - to take a slice of your Event Model and recreate that in Axon Framework. That deserves it's own chapter entirely.

Granted, this is new, of course. We can either do that as part of this PR (which means you need to move the stateful command handler logic into a dedicated page) or in a follow-up PR.

Concluding, that would live the `command-handlers.adoc` devoid of entity-details, which I think is a good thing. For entity-details, we have different pages. That leaves the plain description of Command Handlers super straightforward and to the point, making it an easily digestible topic.

Author: okay i did my best to restructure it and to keep it logical but felt bit difficult

= Command Handlers
:navtitle: Handling

> 
> Command handlers process commands and perform the corresponding business logic.
> A command handler decides whether a requested change is valid and, if so, applies it—by appending events, invoking services, or dispatching further commands.
> 
> [#stateless-command-handlers]

Reviewer 2: Since this is the general page about Command Handlers users would land on this page when using the navigation to look for command handlers. Although i'ts a bit redundant with the information in `index.adoc`, I think it makes sense to put a second paragraph on the intro section here reiterating stateless and stateful, but just briefly stating that stateful handlers are covered in `event-sourced-entity.adoc` and `command-handler-components.adoc` and referencing there.

Then I would also still keep the h2 heading to introduce stateless command handlers.
```suggestion

As explained before, command handlers can be **stateless** (not requiring any prior state) or **stateful** (depending on prior state to enforce business rules).  For an explanation on stateless command handlers, consult xref:#stateless-command-handlers[the section below]. The different patterns for implementing stateful command handlers are covered in more detail in xref:entities/event-sourced-entity.adoc[Event-Sourced Entities] and xref:entities/command-handler-components.adoc[Command Handler Components].

[#stateless-command-handlers]
== Stateless command handler
```

This page explains how to define command handlers, handle commands, and work with the key concepts like `ProcessingContext`, `EventAppender`, and message types.

> In Axon Framework 5, commands are the primary driver of all state changes.

Reviewer: Let's not specifically refer to Axon Framework as "Axon Framework 5" in the main descriptions. If it's migration documentation, it's fine. But otherwise, the docs are versioned already. The fact they're on this page signals it's AF5.

```suggestion
In Axon Framework, commands are the primary driver of all state changes.
```

If a command only performs validation or queries without changing state, consider whether it should be implemented as a query instead.
====

When command handling requires entity state, keep the handler in a separate component and receive the entity as a parameter via `@InjectEntity`.

> See xref:commands:entities/command-handler-components.adoc[Command handler components] for the full pattern, including how it supports Vertical Slice Architecture.

Reviewer 2: could be event-sourced-entity as well
```suggestion
When command handling requires entity state, you can use stateful command handlers as explained in more detail in xref:entities/event-sourced-entity.adoc[Event-Sourced Entities] and xref:entities/command-handler-components.adoc[Command Handler Components].
```

<3> Instance handler receives entities when the parameter is annotated with `@InjectEntity`.
Axon automatically loads it based on the command's routing key.
To load by a different field, see <<_injecting_multiple_entities_with_injectentity,Injecting multiple entities>>.

<4> The handler accesses the entity to make business decision.
This can either be done by invoking entity-specific decision methods or by inspecting the state (as shown in the sample above).

For how to define the entity itself—state fields, event sourcing handlers, `@EntityCreator` patterns, and configuration—see xref:commands:entities/index.adoc[Entities].

[TIP]
====

> The command-centric approach aligns closest with [Vertical Slice-based Architecture](https://www.baeldung.com/java-vertical-slice-architecture). This style is useful when:

Reviewer 2: this refers to the enumeration of the different approaches for stateful commandhandlers ("command-centric approach" vs "entity-centric") that has been removed, would rephrase it a bit
```suggestion
Keeping commands in a separate component and receive the loaded entity as a parameter aligns closest with [Vertical Slice-based Architecture](https://www.baeldung.com/java-vertical-slice-architecture). This style is useful when:
```

[source,java]
----
@CommandHandler

> public String handle(IssueCardCommand cmd, EventAppender eventAppender) { // <1>

Reviewer 2: as a creational handler this would rather be
```suggestion
public static String issue(IssueCardCommand cmd, EventAppender eventAppender) { // <1>
```

* You need payload conversion to a specific type.
* You're integrating with external systems that use different message formats.

> == Configuration

Reviewer: I think the `Configuration` section could do a listing similar to other components.
So we have:

1. Declarative
2. Automated 
3. Automated + Dependency Injection framework

This current block does not show 1. It show 2 at the end, and 3 at the start.

If we could expand this section to show all three, preferably in that order, that would be great!
